sdk/clientcore/core/src/main/java/io/clientcore/core/instrumentation/logging/ExceptionLoggingEvent.java

```diff
@@ -0,0 +1,117 @@
+     * and exception being created.
+     *
+     * @param key String key.
+     * @param value String value.
+     * @return The updated {@link ExceptionLoggingEvent} object.
+     *
+     * @see LoggingEvent#addKeyValue(String, String)
+     */
+    public ExceptionLoggingEvent<T> addKeyValue(String key, String value) {
+        log.addKeyValue(key, value);
+
+        return this;
+    }
+
+    /**
+     * Adds a key with a boolean value to the context of current log
+     * and exception being created.
+     *
+     * @param key Key to associate the provided {@code value} with.
+     * @param value The boolean value.
+     * @return The updated {@link ExceptionLoggingEvent} object.
+     * @see LoggingEvent#addKeyValue(String, boolean)
+     */
+    public ExceptionLoggingEvent<T> addKeyValue(String key, boolean value) {
+        log.addKeyValue(key, value);
+        return this;
+    }
+
+    /**
+     * Adds key with long value to the context of current log event being created.
+     *
+     * @param key Key to associate the provided {@code value} with.
+     * @param value The long value.
+     * @return The updated {@link ExceptionLoggingEvent} object.
+     * @see LoggingEvent#addKeyValue(String, long)
+     */
+    public ExceptionLoggingEvent<T> addKeyValue(String key, long value) {
+        log.addKeyValue(key, value);
+        return this;
+    }
+
+    /**
+     * Sets operation context on the log event being created.
+     * It's used to correlate logs between each other and with other telemetry items.
+     *
+     * @param context operation context.
+     * @return The updated {@link ExceptionLoggingEvent} object.
+     * @see LoggingEvent#setInstrumentationContext
+     */
+    public ExceptionLoggingEvent<T> setInstrumentationContext(InstrumentationContext context) {
+        log.setInstrumentationContext(context);
+        return this;
+    }
+
+    /**
+     * Logs the exception with the provided short message.
+     * The full exception message is enriched with the context of the log event.
+     *
+     * @param shortMessage The short message to log.
+     * @return The created exception.
+     */
+    public T log(String shortMessage) {
+        return log(shortMessage, null);
+    }
+
+    /**
+     * Logs the exception with the provided cause.
+     * The full exception message is enriched with the context of the log event.
+     *
+     * @param cause The cause of the exception.
+     * @return The created exception.
+     */
+    public T log(Throwable cause) {
+        return log(null, cause);
+    }
+
+    /**
+     * Logs the exception with the provided short message and cause.
+     * The full exception message is enriched with the context of the log event.
+     *
+     * @param shortMessage The short message to log.
+     * @param cause The cause of the exception.
+     * @return The created exception.
+     */
+    public T log(String shortMessage, Throwable cause) {
```

Comment:
Would we ever add in support for logging exceptions where we're adding suppressed exceptions as well as causal exceptions?

Reply:
I *think* we don't need anything special here.

I can do something like 

```java
throw LOGGER.throwableAtError((m, c) -> {
                    var ex = new IllegalArgumentException(m, c);
                    ex.addSuppressed(ex);
                    return ex;
                }))
    .addKeyValue("foo", bar)
    .log("Some message");
```

if I want to log exception with suppressed details.

It's not super-convenient, but there are very few places in the code where we add suppressed exceptions today.